pkg/operator/controller/canary/daemonset.go

```diff
@@ -0,0 +1,75 @@
+package canary
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/openshift/cluster-ingress-operator/pkg/manifests"
+	"github.com/openshift/cluster-ingress-operator/pkg/operator/controller"
+
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+)
+
+// ensureCanaryDaemonSet ensures the canary daemonset exists
+func (r *reconciler) ensureCanaryDaemonSet() (bool, *appsv1.DaemonSet, error) {
+	desired := desiredCanaryDaemonSet(r.Config.CanaryImage)
+	haveDepl, current, err := r.currentCanaryDaemonSet()
```

Comment:
`haveDepl` can be a misleading name for a DaemonSet, I suggest using `have` or `haveDs`.

Reply:
Whoops. Fixed.